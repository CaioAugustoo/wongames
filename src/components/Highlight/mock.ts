export default {
  title: 'Read Dead is back!',
  subtitle: 'Come see Johnâ€™s new adventures',
  backgroundImage: '/img/red-dead-img.jpg',
  buttonLabel: 'Buy now',
  buttonLink: '/games/rdr2'
}
