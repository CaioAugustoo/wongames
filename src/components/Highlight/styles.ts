import styled from 'styled-components'

export const Wrapper = styled.main``

export const Title = styled.h2``

export const Subtitle = styled.h3``
