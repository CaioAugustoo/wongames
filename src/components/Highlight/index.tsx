import Button from 'components/Button'
import * as S from './styles'

export type HightLightProps = {
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
}

const Highlight = ({
  title,
  subtitle,
  buttonLabel,
  buttonLink
}: HightLightProps) => (
  <S.Wrapper>
    <S.Title>{title}</S.Title>
    <S.Subtitle>{subtitle}</S.Subtitle>
    <Button as="a" href={buttonLink}>
      {buttonLabel}
    </Button>
  </S.Wrapper>
)

export default Highlight
