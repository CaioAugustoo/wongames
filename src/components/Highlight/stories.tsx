import { Story, Meta } from '@storybook/react/types-6-0'
import Highlight, { HighLightProps } from '.'

import item from './mock'

export default {
  title: 'Highlight',
  component: Highlight,
  args: { ...item }
} as Meta

export const Default: Story<HighLightProps> = (args) => (
  <div style={{ maxWidth: '104rem' }}>
    <Highlight {...args} />
  </div>
)

export const WithFloatImage: Story<HighLightProps> = (args) => (
  <div style={{ maxWidth: '104rem' }}>
    <Highlight {...args} />
  </div>
)

WithFloatImage.args = {
  floatImage: '/img/red-dead-float.png'
}
