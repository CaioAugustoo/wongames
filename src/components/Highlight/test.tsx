import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'

import Highlight from '.'

const props = {
  title: 'Heading 1',
  subtitle: 'Heading 2',
  backgroundImage: '/img/red-dead-img.jpg',
  buttonLabel: 'Buy now',
  buttonLink: '/rdr2'
}

describe('<Highlight />', () => {
  it('should render headings and buttons', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(
      screen.getByRole('heading', { name: /Heading 1/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Heading 2/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /Buy now/i })).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /Buy now/i })).toBeInTheDocument()
  })

  it('should render background image', () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(container.firstChild).toHaveStyle({
      backgroundImage: `url(${props.backgroundImage})`
    })
  })
})
