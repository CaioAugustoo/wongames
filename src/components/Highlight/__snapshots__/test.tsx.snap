// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render headings and buttons 1`] = `
.c5 {
  background: #47c8ff;
  color: #FAFAFA;
  border: none;
  border-radius: 0.4rem;
  padding: 0.8rem;
  cursor: pointer;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c5:hover {
  background: #2499ae;
}

.c0 {
  position: relative;
  height: 23rem;
  display: grid;
  background-image: url(/img/red-dead-img.jpg);
  background-position: center center;
  background-size: cover;
  grid-template-areas: 'floatimage content';
}

.c0::after {
  content: '';
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.6);
}

.c0 .c1 {
  text-align: right;
}

.c2 {
  grid-area: content;
  z-index: 10;
  padding: 1.6rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  font-size: 1.4rem;
  font-weight: 300;
  color: #FAFAFA;
  margin-bottom: 3.2rem;
}

@media (min-width:768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1 c2"
  >
    <h2
      class="c3"
    >
      Heading 1
    </h2>
    <h3
      class="c4"
    >
      Heading 2
    </h3>
    <a
      class="c5"
      href="/rdr2"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</section>
`;
